### PR TITLE
fix(class): keep members in namespace classes

### LINKED ISSUE #4519: namespaced classes lose members if relation statement comes after namespace
### Description

When a relation is declared after a namespace, the members disappear.

The following diagram will not include the `+Id: Guid` in either class
```
    classDiagram
      namespace A {
        class A1 {
          +Id : Guid
        }
      }
      namespace B {
        class B1 {
          +Id: Guid
        }
      }
      A1 --> B1
```
while 
```
    classDiagram
      A1 --> B1
      namespace A {
        class A1 {
          +Id : Guid
        }
      }
      namespace B {
        class B1 {
          +Id: Guid
        }
      }
```
will display the `+Id: Guid` because the relationships are before the namespace declaration.

### Steps to reproduce

See Description and code sample

### Screenshots

mermaid_bug1

### Code Sample

```text
classDiagram
      namespace A {
        class A1 {
          +Id : Guid
        }
      }
      namespace B {
        class B1 {
          +Id: Guid
        }
      }
      A1 --> B1
```

### Setup

- Mermaid version: 3662869ce14be1a64a73fa4e6602f368b6204865 (on `develop` branch)
- Browser and Version: [Chrome, Firefox]

### Additional Context

This is a bug in the parser/classDb file.
Namespaces depend on successfully deleting the class from the base `classes` variable:
https://github.com/mermaid-js/mermaid/blob/3d117815425cbd05105ec0f0192adc0be91931f2/packages/mermaid/src/diagrams/class/classDb.ts#L452

However, when relations are added, they call addClass() (again)
https://github.com/mermaid-js/mermaid/blob/3d117815425cbd05105ec0f0192adc0be91931f2/packages/mermaid/src/diagrams/class/classDb.ts#L132-L136
and so the class is re-instantiated w/o the members that were 'known" from adding to namespaces

### PR DESCRIPTION
**Pull request by tomperr**
## :bookmark_tabs: Summary

Keep members of classes in namespaces, whether relations are defined before or after classes definition

Resolves #4519 

## :straight_ruler: Design Decisions

We keep all classes (including the ones inside a namespace) inside `classes` dictionary so we don't consider them as `undefined` 
 
### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added unit/e2e tests (if appropriate)
- [x] :notebook: have added documentation (if appropriate)
- [x] :bookmark: targeted `develop` branch

### CODE REVIEW

File: packages/mermaid/src/diagrams/class/classRenderer-v2.ts [hunk 1]
```diff
@@ -93,52 +93,54 @@ export const addClasses = function (
...
+    .forEach(function (id) {
+      const vertex = classes[id];
+
+      /**
+       * Variable for storing the classes for the vertex
+       */
+      let cssClassStr = '';
+      if (vertex.cssClasses.length > 0) {
+        cssClassStr = cssClassStr + ' ' + vertex.cssClasses.join(' ');
+      }
```
Comment (sidharthv96) [inline packages/mermaid/src/diagrams/class/classRenderer-v2.ts:107]:
```suggestion
      const cssClassStr = vertex.cssClasses.join(' ');
```

File: packages/mermaid/src/diagrams/class/classRenderer-v2.ts [hunk 2]
```diff
@@ -93,52 +93,54 @@ export const addClasses = function (
...
 
-    if (parent) {
-      g.setParent(vertex.id, parent);
-    }
+      const styles = { labelStyle: '', style: '' }; //getStylesFromArray(vertex.styles);
```
Comment (sidharthv96) [inline packages/mermaid/src/diagrams/class/classRenderer-v2.ts:109]: This is not modified afterwards. Bug?

File: packages/mermaid/src/diagrams/class/classRenderer-v2.ts [hunk 3]
```diff
@@ -93,52 +93,54 @@ export const addClasses = function (
...
+        rx: radius,
+        ry: radius,
+        class: cssClassStr,
+        style: styles.style,
+        id: vertex.id,
+        domId: vertex.domId,
+        tooltip: diagObj.db.getTooltip(vertex.id, parent) || '',
+        haveCallback: vertex.haveCallback,
+        link: vertex.link,
+        width: vertex.type === 'group' ? 500 : undefined,
```
Comment (sidharthv96) [inline packages/mermaid/src/diagrams/class/classRenderer-v2.ts:131]: Avoid magic number 500

**Review by sidharthv96 — CHANGES_REQUESTED** [review]

**Comment by tomperr** [comment]
@sidharthv96 seems like there is a bug on this PR, the only thing I added in `classRenderer-v2.ts` was the filter before the foreach loop

**Comment by tomperr** [comment]
There might be some lines of code in this file that need to be reworked, but I don't think this should be done in this PR (as it is not linked to the issue +  I am not the author of the code you commented)

**Pull request merged**